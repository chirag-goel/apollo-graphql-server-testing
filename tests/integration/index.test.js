
const {
    ApolloServerBase
} = require('apollo-server-core');
const typeDefs = require('@src/typeDefs.js');
const resolvers = require('@src/resolvers.js');
const {
    createTestClient
} = require('apollo-server-testing');
const DataSource = require('@src/dataSource');
describe('Integration Testing With Mocked DataSource', () => {

    const myTestServer = new ApolloServerBase({
        resolvers,
        typeDefs,
    });

    it('QUERY_userData_withSnapshot', async () => {
        const USER_DATA = {
            "chirag": {
                "firstName": "Chirag",
                "lastName": "Goel",
                "website": "http://www.chirag-goel.in"
            }
        };
        DataSource.getUserData = jest.fn(() => USER_DATA);
        const query = '{ userData(userId: "chirag") { firstName lastName website} }';
        const client = createTestClient(myTestServer);
        const res = await client.mutate({
            query
        });
        expect(res.data.userData).toMatchSnapshot();
    });
});