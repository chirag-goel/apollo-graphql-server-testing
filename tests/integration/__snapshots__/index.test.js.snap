// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Integration Testing With Mocked DataSource QUERY_userData_withSnapshot 1`] = `
Object {
  "firstName": "Chirag",
  "lastName": "Goel",
  "website": "http://www.chirag-goel.in",
}
`;
